src/ui.data-list/UDataList.vue

```diff
@@ -1,3 +1,148 @@
+<script setup lang="ts">
+import { computed } from "vue";
+import draggable from "vuedraggable";
+import { merge } from "lodash-es";
+
+import { getDefault } from "../utils/ui.ts";
+import { hasSlotContent } from "../utils/helper.ts";
+
+import UIcon from "../ui.image-icon/UIcon.vue";
+import UEmpty from "../ui.text-empty/UEmpty.vue";
+
+import { UDataList as UDataListName } from "./constants.ts";
+import defaultConfig from "./config.ts";
+import useAttrs from "./useAttrs.ts";
+import { useLocale } from "../composables/useLocale.ts";
+
+import type { UDataListProps, IconSize, DragMoveEvent, ListItem } from "./types.ts";
+
+defineOptions({ inheritAttrs: false });
+
+const props = withDefaults(defineProps<UDataListProps>(), {
+  size: getDefault<UDataListProps>(defaultConfig, UDataListName).size,
+  labelKey: getDefault<UDataListProps>(defaultConfig, UDataListName).labelKey,
+  valueKey: getDefault<UDataListProps>(defaultConfig, UDataListName).valueKey,
+  animationDuration: getDefault<UDataListProps>(defaultConfig, UDataListName).animationDuration,
+  nesting: getDefault<UDataListProps>(defaultConfig, UDataListName).nesting,
+  hideEmptyStateForNesting: false,
+  dataTest: "",
+  config: () => ({}),
+});
+
+const emit = defineEmits([
+  /**
+   * Triggers when item is sorted (after drag).
+   * @property {array} sortData
+   */
+  "dragSort",
+
+  /**
+   * Triggers when edit button is clicked.
+   * @property {number} value
+   * @property {string} label
+   */
+  "clickEdit",
+
+  /**
+   * Triggers when delete button is clicked.
+   * @property {number} value
+   * @property {string} label
+   */
+  "clickDelete",
+]);
+
+defineSlots<{
+  edit: { element: object };
+  delete: { element: object };
+  drag: { element: object };
+  empty: {
+    emptyTitle: string;
+    emptyDescription: string;
+  };
+  label: { item: ListItem; active: boolean };
+  actions: { item: ListItem };
+}>();
+
+const {
+  config,
+  wrapperAttrs,
+  emptyAttrs,
+  draggableAttrs,
+  nestedAttrs,
+  itemWrapperAttrs,
+  itemAttrs,
+  labelAttrs,
+  labelCrossedAttrs,
+  customActionsAttrs,
+  deleteIconAttrs,
+  editIconAttrs,
+  dragIconAttrs,
+} = useAttrs(props);
+const { tm } = useLocale();
+
+const i18nGlobal = tm(UDataListName);
+const currentLocale = computed(() => merge(defaultConfig.i18n, i18nGlobal, props.config.i18n));
+
+const iconSize = computed(() => {
+  const sizes = {
+    sm: "xs",
+    md: "sm",
+    lg: "md",
+  };
+
+  return sizes[props.size] as IconSize;
+});
+
+function isActive(element: { isActive: boolean }) {
```

Comment:
define type for element and use it here. put all known keys in that type, not only isActive